src/ApplicationManager.cpp

```
@@ -133,6 +158,22 @@ void ApplicationManager::processMessages() {
 
+    // Bambu dashboard revert: after scan interruption, return to dashboard
+    if (dashboardRevertAt_ != 0) {
+        uint32_t now = millis();
+        if (now >= dashboardRevertAt_) {
+            dashboardRevertAt_ = 0;
+#ifndef NATIVE_TEST
+            bool dashEnabled = ConfigurationManager::getInstance().isBambuDashboardEnabled();
+#else
+            bool dashEnabled = false;
+#endif
+            if (dashEnabled && trayDashboardState_.has_data && display_) {
```

Clang (14.0.6)

[warning] implicit conversion 'DisplayI *' -> bool

(readability-implicit-bool-conversion)